src/parser/shared/modules/EventHistory.ts

```diff
@@ -229,6 +229,43 @@ class EventHistory extends Module {
     );
     return filteredEvents;
   }
+
+  /**
+   * Get all events of specified type(s) within a time window around a specific timestamp.
+   *
+   * @param eventTypes the event type(s) to get (single EventType or array of EventTypes)
+   * @param timestamp the center point of the time window
+   * @param beforeMs how many milliseconds before the timestamp to search (default: 0)
+   * @param afterMs how many milliseconds after the timestamp to search (default: 0)
+   * @param spell optional spell filter(s)
+   * @returns array of events matching the criteria within the time window
+   */
+  getEventsInTimeWindow<ET extends EventType>(
```

Comment:
the runtime of this is going to be pretty bad i think. probably better to just make `getEvents` understand an array of event types